autofs-5.1.3 - fix strerror_r() parameter declaration in do program_mount()

From: Ian Kent <raven@themaw.net>

Fix an incorrect declaration for an error message buffer on function
program_mount().

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG           |    1 +
 modules/parse_amd.c |    4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

--- autofs-5.0.7.orig/CHANGELOG
+++ autofs-5.0.7/CHANGELOG
@@ -283,6 +283,7 @@
 - add function construct_argv().
 - add amd mount type program mount support.
 - fix memory leak in umount_amd_ext_mount().
+- fix strerror_r() parameter declaration in do program_mount().
 
 25/07/2012 autofs-5.0.7
 =======================
--- autofs-5.0.7.orig/modules/parse_amd.c
+++ autofs-5.0.7/modules/parse_amd.c
@@ -1292,8 +1292,8 @@ static int do_program_mount(struct autof
 	} else {
 		rv = mkdir_path(entry->fs, 0555);
 		if (rv && errno != EEXIST) {
-			char *buf[MAX_ERR_BUF];
-			char * estr;
+			char buf[MAX_ERR_BUF];
+			char *estr;
 
 			estr = strerror_r(errno, buf, MAX_ERR_BUF);
 			error(ap->logopt,
